### PR TITLE
DevWs for Ecolink Intelligent Technology containing containing Ecolink DWZB1-ECO Contact Sensor

### PR DESCRIPTION
**Pull request by marchbold42**

### CODE REVIEW

File: devicetypes/ecolink/ecolink-dwzb1-eco-contact-sensor.src/ecolink-dwzb1-eco-contact-sensor.groovy [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+		capability "Refresh"
+		capability "Health Check"
+		capability "Sensor"
+
+		command "enrollResponse"
+        
+        attribute "tamper", "enum", ["detected", "clear"]
+
+		//fingerprints
+        fingerprint profileId: "0104", deviceId: "0402", inClusters: "0000,0001,0003,0020,0500,0B05", outClusters: "0019", manufacturer: "Ecolink", model: "DWZB1-ECO"//, deviceJoinName: "Ecolink Door/Window"
```
Comment (PKacprowiczS) [inline devicetypes/ecolink/ecolink-dwzb1-eco-contact-sensor.src/ecolink-dwzb1-eco-contact-sensor.groovy]: @marchbold42 I think there's no need for whole new DTH. Could you add fingerprint of your device to `smartsense-open-closed-sensor.groovy`? It should cover its whole functionality.

File: devicetypes/ecolink/ecolink-dwzb1-eco-contact-sensor.src/ecolink-dwzb1-eco-contact-sensor.groovy [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+	//read battery voltage and temperature
+	def refreshCmds = zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0020) +
+			zigbee.readAttribute(zigbee.TEMPERATURE_MEASUREMENT_CLUSTER, 0x0000) +
+            zigbee.readAttribute(zigbee.IAS_ZONE_CLUSTER, 0x0002)
+    
+    return refreshCmds
+}
+
+def configure() {
```
Comment (PKacprowiczS) [inline devicetypes/ecolink/ecolink-dwzb1-eco-contact-sensor.src/ecolink-dwzb1-eco-contact-sensor.groovy]: `smartsense-open-closed-sensor.groovy` has already almost the same configuration as you've implemented. If binding with Poll Cluster is crucial for this model, you can define command for that more easily by utilizing `zigbee.addBinding(<add here Poll Cluster number>)`

**Review by PKacprowiczS — COMMENTED** [review]
I'll ping @tpmanley @varzac for quick review, but I think they gonna agree with my recomendation.

File: devicetypes/ecolink/ecolink-dwzb1-eco-contact-sensor.src/ecolink-dwzb1-eco-contact-sensor.groovy (hunk 1 above)
Comment (varzac) [reply devicetypes/ecolink/ecolink-dwzb1-eco-contact-sensor.src/ecolink-dwzb1-eco-contact-sensor.groovy]: Using an existing DTH is certainly preferable if possible.  It looks like this has some special tamper handling on the ias zone message.  But there is already ecolink related configuring in the other one and it looks like there should still be some ability to reuse that for this device as well.

**Comment by PKacprowiczS** [comment]
@marchbold42 to continue with this request, please provide changes I've asked for on this PR;
- remove `ecolink/ecolink-dwzb1-eco-contact-sensor.src/ecolink-dwzb1-eco-contact-sensor.groovy` file
- add your fingerprint to already existing DTH `smartthings/smartsense-open-closed-sensor.src/smartsense-open-closed-sensor.groovy`
- define `deviceJoinName: "Ecolink Open/Closed Sensor"` in fingerprint
- if necessery, provide changes which are obligatory for device to work correctly in that DTH

**Comment by marchbold42** [comment]
@PKacprowiczS I have performed the changes you request above. Please let me know if there is anything else that I need to do to close this issue. Thank you!

**Comment by PKacprowiczS** [comment]
@marchbold42 if so, you haven't pushed it to your branch

**Comment by marchbold42** [comment]
@PKacprowiczS the changes have now been pushed.

**Comment by shrakuma** [comment]
@greens @PKacprowiczS there is a conflict as i see it . what should be done to resolve the conflict
Thank you @marchbold42  for the chages pushed.

**Comment by greens** [comment]
@marchbold42 you'll need to rebase your branch.

`git rebase --onto upstream/master`

assuming you've already set the upstream repo.

Also we're moving to more generic join names, which are the changes you'll be pulling in. We'd ask that you follow the new naming conventions which are [brand name] [device type].

**Comment by marchbold42** [comment]
@greens @PKacprowiczS @shrakuma I did the rebase and pushed the new commit but it seems that there are still conflicts. What is the best course of action forward for this?

**Comment by greens** [comment]
it looks like you've now just created a new DTH instead of adding a new fingerprint?

Perhaps it would be better to just start over.

**Comment by marchbold42** [comment]
Steven,

What steps should I take to start over?

Thank you,

Mike “Archy” Archbold
Systems Engineer
Ecolink Intelligent Technology, Inc. |  2055 Corte Del Nogal  Carlsbad, CA  92011
PH:  1-855 432-6546 ext. 112 | FAX: 1-760-431-8975
www.DiscoverEcolink.com<http://www.discoverecolink.com/>
[cid:image003.jpg@01D2D95D.DF90F250]
This email and any files transmitted with it are confidential and intended solely for the use of the individual or entity to whom they are addressed. If you have received this email in error please notify the Sender and delete all copies of this email. Please note that any views or opinions presented in this email are solely those of the author and do not necessarily represent those of Ecolink. The recipient should check this email and any attachments for the presence of viruses. Ecolink accepts no liability for any damage caused by any virus transmitted by this email.

**Comment by greens** [comment]
@marchbold42 do a fresh clone of the repository and submit a new PR where you just add your fingerprint to smartsense open/closed sensor

**Comment by PKacprowiczS** [comment]
@marchbold42 let us know if you need any assitance here, it has been a while

**Comment by marchbold42** [comment]
Hello,

I have committed the fingerprint change into the repository on April 22 at https://github.com/marchbold42/SmartThingsPublic/blob/master/devicetypes/smartthings/smartsense-open-closed-sensor.src/smartsense-open-closed-sensor.groovy

This is commit e93c9d51cd8ce3422f46d2660e9e6e93984ff4a5 under parent 856169f.

Please let me know if this change was made in the correct repository branch or if there is something I am missing here. Please let me know where to go from here.

Thank you,

Mike “Archy” Archbold
Systems Engineer
Ecolink Intelligent Technology, Inc. |  2055 Corte Del Nogal  Carlsbad, CA  92011
PH:  1-855 432-6546 ext. 112 | FAX: 1-760-431-8975
www.DiscoverEcolink.com<http://www.discoverecolink.com/>
[cid:image001.jpg@01D659CC.37B09C50]
This email and any files transmitted with it are confidential and intended solely for the use of the individual or entity to whom they are addressed. If you have received this email in error please notify the Sender and delete all copies of this email. Please note that any views or opinions presented in this email are solely those of the author and do not necessarily represent those of Ecolink. The recipient should check this email and any attachments for the presence of viruses. Ecolink accepts no liability for any damage caused by any virus transmitted by this email.

**Comment by greens** [comment]
@marchbold42 so you'll want to make a pull request from your fork to our repo. You can do that by going to https://github.com/SmartThingsCommunity/SmartThingsPublic , selecting the "pull requests" tab, clicking "new pull request", and then "compare across forks". Then specify your fork and branch.

**Comment by marchbold42** [comment]
> @marchbold42 so you'll want to make a pull request from your fork to our repo. You can do that by going to https://github.com/SmartThingsCommunity/SmartThingsPublic , selecting the "pull requests" tab, clicking "new pull request", and then "compare across forks". Then specify your fork and branch.

@greens Got it! Thank you for the instructions. I have submitted the new pull request .

**Comment by greens** [comment]
@marchbold42 Can you tag me on that PR so I can find it?

**Comment by greens** [comment]
@marchbold42 I think you want to `git rm ecolink-zigbee-door-window-sensor.groovy` and commit that, otherwise that file will continue to show up in this PR.

You'll also want to `git rebase upstream/master` and resolve those conflicts.

**Comment by marchbold42** [comment]
@greens I completed the rebase and removed ecolink-zigbee-door-window-sensor.groovy as far as I can tell. Please let me know if there is anything else I need to do.

**Comment by greens** [comment]
@marchbold42 on this PR or a different one?

**Comment by marchbold42** [comment]
@greens I did not make a new pull request as when I go to the pull request screen on github it will only take me here and will not allow me to create a new pull request. What I did was clone the repository on my desktop, did the rebase, made sure the ecolink device handler was not present, and pushed it back up to github.

**Comment by greens** [comment]
@marchbold42 so your local directory and git's reckoning of what files should be in it are different. You still need to call `git rm` on the file to have git forget about it even if you've already deleted it in your OS. You can see in the Files tab above that there are still 2 affected files in this PR.

**Comment by marchbold42** [comment]
@greens Alright I think I have it this time. I appreciate the patience and guidance on this.

**Review by greens — APPROVED** [review]

**Review by greens — CHANGES_REQUESTED** [review]
It looks good in the diff, but the merge dialog is still telling me there are conflicts.

It _might_ work to `git fetch upstream master` and then `rebase` again.

**Comment by marchbold42** [comment]
@greens I attempted the fetch and rebase on my local repository and pushed it back to gitub afterwards. Is the conflict resolved?

**Comment by greens** [comment]
@marchbold42 not sure what the issue is at this point, I've just opened a PR with your new fingerprint here and I'll make sure it gets in: https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/38312

**Comment by marchbold42** [comment]
@greens Thank you for all of your help!

**Comment by greens** [comment]
@marchbold42 I've merged my PR and the fingerprints should be live on production on Tuesday next week.

**Comment by marchbold42** [comment]
@greens Thank you so much!

**Comment by KKlimczukS** [comment]
@greens
Shouldn't this PR be closed since #38312 was merged?
cc: @marchbold42

**Pull request closed without merging**